src/mini_project_ui.py

```diff
@@ -111,7 +111,9 @@ def directory_button_box(self):
 
     def add_directory(self):
         """Function for the button click""" 
-        dir_path, _ = QtWidgets.QFileDialog.getOpenFileName(self, "Select DICOM File",
+        dir_path, _ = QtWidgets.QFileDialog.getOpenFileName(self,
+                                                            caption="Select DICOM File",
+                                                            dir=str(pathlib.Path(self.path).parent),
```

Comment:
Is self.path ever None, empty string, or have no parent?
I think some kind of guard is needed here